src/mlpack/tests/image_test.cpp

```diff
@@ -399,3 +399,41 @@ TEMPLATE_TEST_CASE("ResizeCropUpscaleTest", "[ImageTest]", unsigned char,
   REQUIRE(info.Channels() == inputChannels);
   REQUIRE(image.n_elem == info.Height() * info.Width() * info.Channels());
 }
+
+/**
+ * Test that converts stb image layout to mlpack layout.
+ */
+TEST_CASE("FromSTBLayout", "[ImageTest]")
+{
+  arma::mat image = arma::regspace(0, 26);
+  data::ImageInfo info(3, 3, 3);
+
+  arma::mat newLayout = ImageLayout(image, info);
+
+  std::vector<double> expectedOutput = {
+    0, 3, 6, 9, 12, 15, 18, 21, 24,
+    1, 4, 7, 10, 13, 16, 19, 22, 25,
+    2, 5, 8, 11, 14, 17, 20, 23, 26,
+  };
+
+  arma::mat expectedImage(expectedOutput);
+  CheckMatrices(newLayout, expectedImage);
+}
+
+/**
+ * Test that converts mlpack image layout to stb layout.
+ */
+TEST_CASE("ToSTBLayout", "[ImageTest]")
+{
+  data::ImageInfo info(3, 3, 3);
+  std::vector<double> input = {
+    0, 3, 6, 9, 12, 15, 18, 21, 24,
+    1, 4, 7, 10, 13, 16, 19, 22, 25,
+    2, 5, 8, 11, 14, 17, 20, 23, 26,
+  };
+  arma::mat inputMat(input);
+
+  arma::mat newLayout = STBLayout(inputMat, info);
+  arma::mat expectedImage = arma::regspace(0, 26);
+  CheckMatrices(newLayout, expectedImage);
+}
```

Comment:
I'd also consider adding some tests for:

 * when the image has only one channel
 * when the image has only one pixel
 * when there are multiple images
 * and when the image matrix is empty :)